core/src/Kokkos_Graph.hpp

```diff
@@ -161,6 +161,12 @@ Graph<ExecutionSpace> create_graph(Closure&& arg_closure) {
 #include <impl/Kokkos_GraphNodeImpl.hpp>
 #include <impl/Kokkos_Default_Graph_Impl.hpp>
 #include <Cuda/Kokkos_Cuda_Graph_Impl.hpp>
+#if defined(KOKKOS_ENABLE_HIP)
+// The implementation of hipGraph in ROCm 5.2 is bugged, so we cannot use it.
+#if !((HIP_VERSION_MAJOR == 5) && (HIP_VERSION_MINOR == 2))
```

Comment:
the PR description says that starting from 5.2 it is supported, but then here you check for not 5.2. 
Can we just guard for HIP > 5.2 ? (then i read Damien's comment, i agree on adding a comment to clarify the min version)

Reply:
That's not what the description says, hipGraph was introduced in ROCm 4.5. ROCm 5.2 is the oldest version of ROCm that we support  but unfortunately that in ROCm 5.2m hipGraph is bugged (from what I read older versions worked)

Reply:
sorry i misread the 4.5 in the description for a 4.2! 
I still think that it would be helpful to add to the comment the fact that we are assuming the ROCm minimum version to be 5.2. which i think right now it is implicitly assumed here right?

Reply:
> I still think that it would be helpful to add to the comment the fact that we are assuming the ROCm minimum version to be 5.2. which i think right now it is implicitly assumed here right?

I don't understand what you mean. Among the versions of ROCm that we support, one of them is bugged and therefore, for that bugged version, we use the fallback implementation. Why does it matter that it's the minimal version that we support?

Reply:
because that guard has no base to exist if the minim version is > 5.2. so we could at least say that this is here becaause at the time of this development the min version kokkos supports is 5.2 and should be removed when the min gets updated. so that 2 years from now when the min will be 6 or whatever, anybody reading this will realize this is not just a forgotten thing

Reply:
if you want to take my suggestion great, otherwise no big deal ! i can move along